src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -362,5 +362,47 @@ BuildCommand BuildProject(string buildNumber)
                 return command;
             }
         }
+
+        [Theory]
+        [InlineData(false, false, false)]
+        [InlineData(true, false, true)]
+        [InlineData(false, true, true)]
+        [InlineData(true, true, true)]
+        public void GenerateUserSecrets(bool referenceAspNetCore, bool referenceExtensionsUserSecrets, bool shouldHaveAttribute)
+        {
+            var testProject = new TestProject()
+            {
+                Name = "UserSecretTest",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp3.0"
+            };
+
+            testProject.AdditionalProperties["UserSecretsId"] = "SecretsIdValue";
+
+            if (referenceAspNetCore)
+            {
+                testProject.FrameworkReferences.Add("Microsoft.AspNetCore.App");
+            }
+            if (referenceExtensionsUserSecrets)
+            {
+                testProject.PackageReferences.Add(new TestPackageReference("Microsoft.Extensions.Configuration.UserSecrets", "3.0.0"));
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: referenceAspNetCore.ToString() + referenceExtensionsUserSecrets.ToString())
+                .Restore(Log, testProject.Name);
+
+            var buildCommand = new BuildCommand(Log, testAsset.TestRoot, testProject.Name);
+
+            buildCommand.Execute()
+                .Should()
+                .Pass();
+
+            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory(testProject.TargetFrameworks).FullName, testProject.Name + ".dll");
+
+            if (shouldHaveAttribute)
```

Comment:
What about testing it's not there when it shouldn't be there? That's the bug being fixed.

Reply:
The bug being fixed is that currently the build will fail, because the attribute is emitted but the assembly it's from isn't referenced.  But we can still check that the attribute isn't emitted.

Reply:
Right, what I'm saying is that I don't see any assert in the test for the parameters `[InlineData(false, false, false)]` since line 402 will be false.

Reply:
`buildCommand.Execute().Should().Pass()` is asserted in all cases.

Reply:
Cool, I wasn't so familiar with this assertion style.